packages/vm/lib/evm/evm.ts

```diff
@@ -424,16 +425,16 @@ export default class EVM {
     }
   }
 
-  async _generateAddress(message: Message): Promise<Buffer> {
+  async _generateAddress(message: Message): Promise<Address> {
     let addr
     if (message.salt) {
-      addr = generateAddress2(message.caller, message.salt, message.code as Buffer)
+      addr = generateAddress2(message.caller.buf, message.salt, message.code as Buffer)
     } else {
       const acc = await this._state.getAccount(message.caller)
       const newNonce = acc.nonce.subn(1)
-      addr = generateAddress(message.caller, newNonce.toArrayLike(Buffer))
+      addr = generateAddress(message.caller.buf, newNonce.toArrayLike(Buffer))
```

Comment:
Can we make `generateAddress` and `generateAddress2` also take `Address` as input type?

Reply:
unfortunately those have been in `ethereumjs-util` for a while so we'd have to make breaking changes there. it would be nice if they output an `Address` as well.

another thing I noticed is it would be better if it accepted nonce as a `BN` instead of `Buffer`.

Reply:
Ah woops, thought these were defined inside EVM